lib/matplotlib/axes/_axes.py

```diff
@@ -3050,11 +3050,42 @@ def boxplot(self, x, notch=False, sym=None, vert=True, whis=1.5,
 
         .. plot:: mpl_examples/statistics/boxplot_demo.py
         """
+        if not rcParams['boxplot.whiskers'] == 1.5:
+            whis = rcParams['boxplot.whiskers']
+        if rcParams['boxplot.bootstrap']:
+            bootstrap = rcParams['boxplot.bootstrap']
         bxpstats = cbook.boxplot_stats(x, whis=whis, bootstrap=bootstrap,
                                        labels=labels)
-        # make sure we have a dictionary
-        if flierprops is None:
-            flierprops = dict()
+
+        def _update_dict(dictionnary, rc_name, properties):
+            """ Loads properties in the dictionnary from rc file if not already
+            in the dictionnary"""
+            rc_str = 'boxplot.{0}.{1}'
+            if dictionnary is None:
+                dictionnary = dict()
+            for prop in properties:
+                if prop not in dictionnary:
```

Comment:
you can do this whole block is

```python
dictionnary.setdefault(prop, rcParams[rc_str.format(rc_name, prop))
```

Also, please spell your variable names right.  I would all this as `prop_dict` or something like that.